Sources/Common/Core/Math/index.js

```diff
@@ -609,24 +609,31 @@ export function multiply3x3_mat3(a_3x3, b_3x3, out_3x3) {
   }
 }
 
-export function multiplyMatrix(a, b, rowA, colA, rowB, colB, out_rowXcol) {
+export function multiplyMatrix(a, b, rowA, colA, rowB, colB, out_rowA_colB) {
   // we need colA == rowB
   if (colA !== rowB) {
     vtkErrorMacro('Number of columns of A must match number of rows of B.');
   }
 
   // If a or b is used to store the result, copying them is required
-  const copyA = [...a];
-  const copyB = [...b];
+  const copyA =
+    a === out_rowA_colB || (a.buffer && a.buffer === out_rowA_colB.buffer)
```

Comment:
why `a === out_rowA_colB` is not enough ?

Reply:
Because you can create two typed arrays that are different but use the same buffer:
```js
const x = new Float32Array(1);
const y = new Float32Array(x.buffer);

x[0] = 42;
console.log(x); // [42]
y[0] = 3.14;
console.log(x); // [3.14]

console.log(x === y); // false
console.log(x.buffer === y.buffer); // true
```

Reply:
Fair enough, but if you are sharing the same buffer in 2 different arrays and passing those arrays to `multiplyMatrix`, you seem ok to take the risk that `a` is not copied (if it is correctly documented).

Or said differently, here is an example where `a` and `b` do not have to be copied that could make sense:
```
const inputsOutputs = new Float32Array(10000);
...
multiplyMatrix(inputsOutputs.subArray(0), inputsOutputs.subArray(9), 3,3, inputsOutputs.subArray(18));
```

And if it is an issue for the caller, then he can still do the `copy` him/herself before calling `multiplyMatrix`. Preventing the copy is not possible with your implementation

Reply:
Yes, it is a limitation of this implementation, it sometimes makes copies that are not needed.
I just wanted to avoid some copies but keep the old API.
I could you `byteOffset` in case of TypedArray, but isn't it overengineering?

Reply:
I suggest to keep the test light: `const copyA = a === out_rowA_colB ? [...a] : a;`

Reply:
I reverted and created a new issue